### PR TITLE
[jsk_pcl_ros] Convert cluster point indices to mask and label image

### PR DESCRIPTION
**Pull request by wkentaro**
cpi_to_mask_image
cpi_to_label

### CODE REVIEW

**Comment by garaemon** [comment]
only for organized pointcloud?

File: jsk_pcl_ros/src/cluster_point_indices_decomposer_nodelet.cpp [hunk 1]
```diff
@@ -361,6 +367,14 @@ namespace jsk_pcl_ros
                                                input->header.frame_id,
                                                tf_prefix_ + (boost::format("output%02u") % (i)).str()));
       }
+      // create mask & label image from cluster indices
+      for (size_t j = 0; j < sorted_indices[i]->size(); j++) {
+        int index = sorted_indices[i]->data()[j];
+        int width_index = index % input->width;
+        int height_index = index / input->width;
+        mask.at<uchar>(height_index, width_index) = 255;
+        label.at<int>(height_index, width_index) = (int)i;
```
Comment (garaemon) [inline jsk_pcl_ros/src/cluster_point_indices_decomposer_nodelet.cpp]: 0 should be avoided

**Comment by wkentaro** [comment]
> only for organized pointcloud?

That only use point indices, so I think organized/unorganized is not important.
organized/unorganized is important when we process point cloud, right?

File: jsk_pcl_ros/src/cluster_point_indices_decomposer_nodelet.cpp (hunk 1 above)
Comment (wkentaro) [inline jsk_pcl_ros/src/cluster_point_indices_decomposer_nodelet.cpp]: > 0 should be avoided

I will update. thanks.

**Comment by wkentaro** [comment]
I updated the commit.

**Comment by garaemon** [comment]
conversion to label image only works for image-like organized pointcloud

need to verify it

2015年10月15日木曜日、Kentaro Wada<notifications@github.com>さんは書きました:

> I updated the commit.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/1231#issuecomment-148132470>
> .
>


-- 
✉︎ from iPhone

**Comment by wkentaro** [comment]
> conversion to label image only works for image-like organized pointcloud

It uses input cloud's width and height value.
https://github.com/jsk-ros-pkg/jsk_recognition/pull/1231/files#diff-e9363841a8ce4af3c4e097a96239ef4fR341

If it is NULL (maybe the pointcloud sensed by laser has NULL), the label and mask should be disabled.
How can I verify it? Should I check if it is `int` or are there some special value?
Actually I don't know what kind of value is there when sensed with laser.

**Comment by garaemon** [comment]
non organized pointcloud has hesght=1 and generated images should be invalid

**Comment by wkentaro** [comment]
Ok. I will skip creating mask & label image when height is 1.

**Comment by wkentaro** [comment]
@garaemon 
is this change ok? 
https://github.com/wkentaro/jsk_recognition/commit/e11e36a3302560e0c7ee907f0966766dee2d2aa8?w=1

**Comment by garaemon** [comment]
looks good

2015年10月15日木曜日、Kentaro Wada<notifications@github.com>さんは書きました:

> @garaemon <https://github.com/garaemon>
> is this change ok?
> wkentaro@e11e36a?w=1
> <https://github.com/wkentaro/jsk_recognition/commit/e11e36a3302560e0c7ee907f0966766dee2d2aa8?w=1>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/1231#issuecomment-148269489>
> .
>


-- 
✉︎ from iPhone

**Comment by wkentaro** [comment]
I squashed the commits.
Please review after the test passes.

**Pull request merged**